index.js

```diff
@@ -1,4 +1,41 @@
-export default function uniqueRandom(minimum, maximum) {
+function _exhaustive(minimum, maximum) {
+	function integersInRange(minimum, maximum) {
+		const array = [];
+		for (let index = minimum; index < maximum + 1; index++) {
+			array.push(index);
+		}
+
+		return array;
+	}
+
+	let restorer = [];
+	let toExtract = integersInRange(minimum, maximum);
+	let previousValue;
+
+	return function random() {
+		if (toExtract.length === 0) {
+			[toExtract, restorer] = [restorer, toExtract];
+		}
+
+		const value = toExtract[Math.floor(Math.random() * toExtract.length)];
+		if (previousValue === value) {
+			return random();
+		}
+
+		toExtract = toExtract.filter(item => item !== value);
```

Comment:
I'd rather have the array spliced at the index instead of being filtered for perf

Reply:
Just updated